pkg/cmd/registry/artifact/crud/list/list.go

```diff
@@ -0,0 +1,210 @@
+
+	"gopkg.in/yaml.v2"
+)
+
+// row is the details of a Service Registry instance needed to print to a table
+type artifactRow struct {
+	// The ID of a single artifact.
+	Id string `json:"id" header:"ID"`
+
+	Name string `json:"name,omitempty" header:"Name"`
+
+	CreatedOn string `json:"createdOn" header:"Created on"`
+
+	CreatedBy string `json:"createdBy" header:"Created By"`
+
+	Type registryinstanceclient.ArtifactType `json:"type" header:"Type"`
+
+	State registryinstanceclient.ArtifactState `json:"state" header:"State"`
+}
+
+type Options struct {
+	group string
+
+	registryID   string
+	outputFormat string
+
+	page  int32
+	limit int32
+
+	IO         *iostreams.IOStreams
+	Config     config.IConfig
+	Connection factory.ConnectionFunc
+	Logger     func() (logging.Logger, error)
+	localizer  localize.Localizer
+}
+
+// NewListCommand creates a new command for listing registry artifacts.
+func NewListCommand(f *factory.Factory) *cobra.Command {
+	opts := &Options{
+		Config:     f.Config,
+		Connection: f.Connection,
+		Logger:     f.Logger,
+		IO:         f.IOStreams,
+		localizer:  f.Localizer,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "list",
+		Short: "List artifacts",
+		Long:  "List all artifacts for the group by specified output format (by default table)",
+		Example: `
+## List all artifacts for the default artifacts group
+rhoas service-registry artifact list
+
+## List all artifacts with explicit group 
+rhoas service-registry artifact list --group=my-group
+
+## List all artifacts with limit and group
+rhoas service-registry artifact list --page=2 --limit=10
+		`,
+		Args: cobra.RangeArgs(0, 1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if opts.outputFormat != "" && !flagutil.IsValidInput(opts.outputFormat, flagutil.ValidOutputFormats...) {
+				return flag.InvalidValueError("output", opts.outputFormat, flagutil.ValidOutputFormats...)
+			}
+
+			if len(args) > 0 {
+				opts.group = args[0]
+			}
+
+			if opts.page < 1 || opts.limit < 1 {
+				return errors.New("page and limit values should be bigger than 1")
+			}
+
+			if opts.registryID != "" {
+				return runList(opts)
+			}
+
+			cfg, err := opts.Config.Load()
+			if err != nil {
+				return err
+			}
+
+			if !cfg.HasServiceRegistry() {
+				return errors.New("No service Registry selected. Use rhoas registry use to select your registry")
+			}
+
+			opts.registryID = fmt.Sprint(cfg.Services.ServiceRegistry.InstanceID)
+
+			return runList(opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.group, "group", "g", "", "Artifact group")
+
+	cmd.Flags().Int32VarP(&opts.page, "page", "", 1, "Page number")
+	cmd.Flags().Int32VarP(&opts.limit, "limit", "", 100, "Page limit")
+
+	cmd.Flags().StringVarP(&opts.registryID, "instance-id", "", "", "Id of the registry to be used. By default uses currently selected registry")
+	cmd.Flags().StringVarP(&opts.outputFormat, "output", "o", "", "Output format (json, yaml, yml, table)")
```

Comment:
```suggestion
	cmd.Flags().StringVarP(&opts.outputFormat, "output", "o", "", "Output format (json, yaml, yml)")
```

Table is not an option.